velox/functions/sparksql/tests/ArrayMaxTest.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <type/Timestamp.h>
+#include <limits>
+#include <optional>
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+class ArrayMaxTest : public SparkFunctionBaseTest {
+ protected:
+  template <typename T>
+  void testExpr(
+      const std::optional<T>& expected,
+      const std::vector<std::optional<T>>& input) {
+    std::vector<std::vector<std::optional<T>>> vec = {input};
+    auto row = makeRowVector({makeNullableArrayVector(vec)});
+    auto result = evaluateOnce<T>("array_max(C0)", row);
+    EXPECT_EQ(expected, result);
+  }
+};
+
+TEST_F(ArrayMaxTest, boolean) {
+  testExpr<bool>(true, {true, false});
```

Comment:
It would be better to move EXPECT_EQ out of testExpr into here. This way when test fails, you'll know which test case failed. Also, consider renaming testExpr to arrayMax for clarity.

```
EXPECT_EQ(arrayMax<bool>({true, false}), true);
```

Reply:
done

Reply:
It is weird after change like this. some test failed
```cpp
../../velox/functions/sparksql/tests/ArrayMinTest.cpp:157: Failure
Expected equality of these values:
  arrayMin<T>({-0.00001, -0.0002, 0.0001})
    Which is: (-0.0002)
  -0.0002
../../velox/functions/sparksql/tests/ArrayMinTest.cpp:158: Failure
Expected equality of these values:
  arrayMin<T>({-0.0001, -0.0002, kMax, kNaN})
    Which is: (-0.0002)
  -0.0002
```
@PHILO-HE Could you take some time to address this? Sorry for no more bandwidth for this.